### PR TITLE
Automation using PyAutoGui - Keyboard and Message Box functionality

### PR DESCRIPTION
**Pull request by manndthakkar**
I have added two new tutorials, covering keyboard controls and message box functions using PyAutoGui. These tutorials provide comprehensive guidance on interacting with keyboards and displaying message boxes in Python scripts, enhancing automation capabilities and user interactivity.

Changes:
- Added "Keyboard Controls" tutorial, covering writing text, pressing individual keys, simultaneous key presses, multiple key presses, hold() context manager, and hotkey() function.
- Included "Message Box Functions" tutorial, detailing the alert(), confirm(), prompt(), and password() functions for displaying message boxes using PyMsgBox.

These tutorials aim to help CSC301 students to incorporate versatile interactions and seamless automation tasks within their Python scripts.

This pull request is ready for review and merging into the main branch upon reviewing.

### CODE REVIEW

File: Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md [hunk 1]
```diff
@@ -166,7 +166,84 @@ pyautogui.doubleClick()
 ![sample_gif](./assets/move_folder.gif)
 
 ## Keyboard Controls
-Pending
+## Keyboard Controls
```
Comment (manav393) [inline Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md:169]: It seems you have accidentally added another heading for the keyboard controls function.

File: Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md [hunk 2]
```diff
@@ -166,7 +166,84 @@ pyautogui.doubleClick()
...
+Once familiar with mouse functions, exploring keyboard interactions becomes the next vital step. PyAutoGui offers various methods for keyboard input, enhancing automation capabilities.
```
Comment (manav393) [inline Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md:170]: The brief introduction is great here. Consider adding a couple of example use cases in order to provide motivation behind the need to use the keyboard functionality.

File: Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md [hunk 3]
```diff
@@ -166,7 +166,84 @@ pyautogui.doubleClick()
...
+
+```python
+>>> with pyautogui.hold('ctrl'):
+        pyautogui.press(['tab', 'tab', 'tab'])
+```
+
+The hotkey() Function: To execute hotkeys or keyboard shortcuts efficiently, utilize the hotkey() function:
+
+```python
+>>> pyautogui.hotkey('ctrl', 'shift', 'p')   # Executes Ctrl + Shift + tab
```
Comment (manav393) [inline Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md:211]: There is a minor error here; the comment should say 'Executes Ctrl + Shift + p'.

File: Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md [hunk 4]
```diff
@@ -166,7 +166,84 @@ pyautogui.doubleClick()
...
+Shows a message box with text input field, along with OK and Cancel buttons. Users can input text, and it returns the entered text or None if Cancel was clicked.
+
+### The password Function
+
+```python
+pyautogui.password(text='', title='', default='', mask='*')
+```
+Similar to the `prompt()` function, it displays a message box with a text input field. However, the typed characters are masked, typically with asterisks (*). It returns the entered text or None if Cancel was clicked.
+
+These functions empower users to handle various types of user interactions effectively within their Python scripts, enhancing the user experience and interactivity of their applications.
```
Comment (manav393) [inline Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md:249]: The best part about this section is that it is concise and straight to the point, allowing readers to integrate it into their use easily. However, this section as a whole would benefit from some pictures to give the reader an idea of what these popups may look like. I would also recommend adding more details about the respective arguments that each message box function takes and how the user can leverage them.

File: Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md [hunk 5]
```diff
@@ -166,7 +166,84 @@ pyautogui.doubleClick()
...
+>>> with pyautogui.hold('ctrl'):
+        pyautogui.press(['tab', 'tab', 'tab'])
+```
+
+The hotkey() Function: To execute hotkeys or keyboard shortcuts efficiently, utilize the hotkey() function:
+
+```python
+>>> pyautogui.hotkey('ctrl', 'shift', 'p')   # Executes Ctrl + Shift + tab
+```
+These functionalities empower users with versatile keyboard interactions, facilitating seamless automation tasks.
```
Comment (manav393) [inline Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md:213]: Overall, this section is really well done. It wastes no time and is super easy to read.

File: Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md [hunk 6]
```diff
@@ -166,7 +166,84 @@ pyautogui.doubleClick()
...
 ## Keyboard Controls
-Pending
+## Keyboard Controls
+Once familiar with mouse functions, exploring keyboard interactions becomes the next vital step. PyAutoGui offers various methods for keyboard input, enhancing automation capabilities.
+
+We can write something using the `write()` keyword. This will simply type the argument that is passed into it, which is a string.
+
+```python
+>>> pyautogui.write('Hi!')                  # Prints 'Hi'
+>>> pyautogui.write('Hi!', interval=0.25)   # Similar to the move function, the second argument is
```
Comment (manav393) [inline Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md:176]: Great example, Mann! Could you please add some pictures/gifs to demonstrate an example? Another addition to consider is keyboard presses while the program is running. For example, what will happen if I press the caps lock button on my keyboard mid-execution?

**Review by manav393 — CHANGES_REQUESTED** [review]
This is a great addition to the PyAutoGUI tutorial. It provides clear and concise explanations of various keyboard interaction methods and message box functions in PyAutoGui. Each function is explained with examples, making it easy for readers to understand how to use them. Some use-case examples and pictures would round it off perfectly. Please look at my comments where I have put in some suggestions as well as pointed out some minor errors.

File: Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md (hunk 1 above)
Comment (manndthakkar) [reply Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md:169]: Yes, will remove the first heading.

File: Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md (hunk 2 above)
Comment (manndthakkar) [reply Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md:170]: Agreed! Thats a great idea. I will add use cases catered towards CSC301 students.

File: Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md (hunk 6 above)
Comment (manndthakkar) [reply Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md:176]: Sure, will add visuals to support the tutorial.

File: Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md (hunk 3 above)
Comment (manndthakkar) [reply Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md:211]: Great catch.

File: Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md (hunk 5 above)
Comment (manndthakkar) [reply Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md:213]: Thanks Manav!

File: Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md (hunk 4 above)
Comment (manndthakkar) [reply Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md:249]: I agree. Will add images showing the output of each function.

**Review by manav393 — APPROVED** [review]
Great job Mann! The documentation is now comprehensive and ready to merge, providing valuable insights into keyboard controls and message box functions in PyAutoGui.

**Pull request merged**